libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationRouteProcessor.kt

```diff
@@ -0,0 +1,141 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.LegStep
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.geojson.Geometry
+import com.mapbox.geojson.Point
+import com.mapbox.navigator.NavigationStatus
+import com.mapbox.navigator.RouteState
+import com.mapbox.services.android.navigation.v5.routeprogress.CurrentLegAnnotation
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgressStateMap
+
+internal class NavigationRouteProcessor {
+    private val ONE_INDEX = 1
+    private val ONE_SECOND_IN_MILLISECONDS = 1000.0
+    private val FIRST_BANNER_INSTRUCTION = 0
+
+    private val progressStateMap = RouteProgressStateMap()
+    private var previousRouteProgress: RouteProgress? = null
+    private var previousStatus: NavigationStatus? = null
+    private var route: DirectionsRoute? = null
```

Comment:
Can we make this non-nullable? The function `buildNewRouteProgress` now returns a nullable `route`. Looking at the file `RouteProcessorRunnable/Line65` that calls `buildNewRouteProgress` does not accept a nullable route. I am not sure how the app will behave if we pass a null route down the application

Reply:
👍

Reply:
The return value of buildNewRouteProgress() is checked in the converted  RouteProcessorRunnable.kt. Null safety is maintained